bambou/session.go

```diff
@@ -196,15 +219,24 @@ func (s *Session) send(request *http.Request, info *FetchingInfo) (*http.Respons
 		request, _ = http.NewRequest(request.Method, newURL, request.Body)
 		return s.send(request, info)
 
-	case http.StatusConflict:
-		berr := NewError(response.StatusCode, "")
-		if err := json.NewDecoder(response.Body).Decode(&berr); err != nil {
-			return nil, NewError(ErrorCodeJSONCannotDecode, err.Error())
+	case http.StatusConflict, http.StatusNotFound:
+		var vsdresp VsdErrorList
+
+		body, _ := ioutil.ReadAll(response.Body)
+		log.Debugf("Response Body: %s", string(body))
+
+		if err := json.Unmarshal(body, &vsdresp); err != nil {
+			return nil, NewBambouError("", err.Error())
+		}
+		// Check if there is an _actual_ VSD response -- we may get a bogus 40x from e.g. tests
+		if len(vsdresp.VsdErrors) == 0 {
+			return nil, NewBambouError("", response.Status)
```

Comment:
Why are we not returning the status code? This could help determining the server side issue